### PR TITLE
Bugfix/various vfs fixes

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

File: src/libsync/vfs/cfapi/cfapiwrapper.cpp [hunk 1]
```diff
@@ -618,7 +619,6 @@ OCC::CfApiWrapper::PlaceHolderInfo OCC::CfApiWrapper::findPlaceholderInfo(const
     if (result == S_OK) {
         return info;
     } else {
-        qCWarning(lcCfApiWrapper()) << "Couldn't get placeholder info" << QString::fromWCharArray(_com_error(result).ErrorMessage());
```
Comment (allexzander) [inline src/libsync/vfs/cfapi/cfapiwrapper.cpp:621]: @mgallien Why remove this? I thought we are still in the process of improving the VFS and fixing issues. These logs might be useful.

File: src/libsync/vfs/cfapi/cfapiwrapper.cpp [hunk 2]
```diff
@@ -586,13 +586,14 @@ OCC::CfApiWrapper::FileHandle OCC::CfApiWrapper::handleForPath(const QString &pa
...
-    if (QFileInfo(path).isDir()) {
+    QFileInfo pathFileInfo(path);
+    if (pathFileInfo.exists() && pathFileInfo.isDir()) {
         HANDLE handle = nullptr;
         const qint64 openResult = CfOpenFileWithOplock(path.toStdWString().data(), CF_OPEN_FILE_FLAG_NONE, &handle);
         if (openResult == S_OK) {
             return {handle, [](HANDLE h) { CfCloseHandle(h); }};
         }
-    } else {
+    } else if (pathFileInfo.exists() && pathFileInfo.isFile()) {
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/libsync/vfs/cfapi/cfapiwrapper.cpp:596]: sure, that would be much better

File: src/libsync/vfs/cfapi/cfapiwrapper.cpp (hunk 1 above)
Comment (mgallien) [reply src/libsync/vfs/cfapi/cfapiwrapper.cpp:621]: many time `OCC::CfApiWrapper::findPlaceholderInfo` is called to see if the info exists and the result is then check by the caller that will also produce a log output
we then gets two log lines or we get this one but the case is handled by the code
having `result != S_OK` is not an error case and the code is expecting it
so the log is being produced but just create noise

**Review by allexzander — APPROVED** [review]
Nice. See if you need a squash

**Review by camilasan — APPROVED** [review]

**Pull request merged**